src/api_server/src/lib.rs

```diff
@@ -3,6 +3,7 @@
 
 #![deny(missing_docs)]
 #![warn(clippy::ptr_as_ptr)]
+#![warn(clippy::cast_lossless)]
```

Comment:
Warning on this error from inside each crate is polluting the code too much (even more so if we decide to warn on multiple cast related errors).
Apart from polluting the code, this method of putting the warn cfg inside each crate is not enforcing it on future crates, right?

I see you can directly specify to the `cargo clippy` command the config we want to warn on. Can you give it a try?

Reply:
I think for testing this approach makes sense, however in development it can become awkward to need to enter many command line arguments rather than just run `cargo clippy`. It might be worth including the command line arguments in the testing, but I would still argue for these tags in the code.

What do you think?

Reply:
Let s try out with a clippy.toml, at least the ones we are interested in are there: https://rust-lang.github.io/rust-clippy/master/index.html.

Reply:
@StemCll In this case when using a `clippy.toml` file it may make sense to open a new issue and pr on the topic of replacing all the crate level `#![warn()]` statements with `clippy.toml`.

Reply:
Unfortunately `clippy.toml` is not a place where one can deny / allow lints ([source](https://github.com/rust-lang/rust-clippy#configure-the-behavior-of-some-lints). Here's an issue relating to the exact case as we have here -> https://github.com/rust-lang/rust-clippy/issues/1313

Reading through your conversation, I would keep warns in one place - allow only specific functions / mods where needed. One of the reasons here is that there are more lints to come (waiting for your comments [here](https://github.com/firecracker-microvm/firecracker/pull/3237) :smile: ) and in each lib.rs file we will have more and more warns. It's not bad, but it would be easier to maintain to keep all warns in one place.

In terms of development - I don't know how you work with code, but I'd just extend the capabilities of the `devtool` script to run clippy check without Pytest. Pytest output is less readable for me, thus I prefer to run clippy without it.

Reply:
another alternative is to specify them in the clippy command; I posted an example of this here: https://github.com/firecracker-microvm/firecracker/pull/3258

Reply:
Alright, this's some solution. What should I do then?

Reply:
hi! I synced offline with @JonathanWoollett-Light and decided to go with the current form (of specifying clippy in source code) for the time being!

You can go ahead and rebase your PR, thanks!

Reply:
@dianpopa consider that for the moment it is fair to go with an implementation here using `#[warn(clippy::)]` rather than the command line.

So if you can rebase with this approach that be would great.

Reply:
done :heavy_check_mark: